firstapp/app/src/androidTest/java/com/example/firstapp/CUJExecutor.java

```diff
@@ -0,0 +1,40 @@
+package com.example.firstapp;
+import android.os.SystemClock;
+import androidx.test.uiautomator.UiDevice;
+import androidx.test.uiautomator.UiObject;
+import org.junit.Test;
+
+public class CUJExecutor {
+
+  
+
+    /**
+     * Executes a CUJ with preparatory actions pre and measured actions post. Caches information for later use.
+     */
+
+    public static void executeCUJ(String[] preCUJ, String[] postCUJ) throws Exception {
+        String[] CUJ = new String[preCUJ.length + postCUJ.length];
+        if (preCUJ != null) {
```

Comment:
To make the code simpler, the preCUJ should never be null.  Instead, an empty array should be passed in instead.